src/include/daos_fs.h

```diff
@@ -1021,6 +1021,24 @@ dfs_osetattr(dfs_t *dfs, dfs_obj_t *obj, struct stat *stbuf, int flags);
 int
 dfs_access(dfs_t *dfs, dfs_obj_t *parent, const char *name, int mask);
 
+/**
+ * Check access permissions on an object. Symlinks are not dereferenced if
+ * O_NOFOLLOW is set in flag. Otherwise, it is same as dfs_access.
+ *
+ * \param[in]	dfs	Pointer to the mounted file system.
+ * \param[in]	parent	Opened parent directory object. If NULL, use root obj.
+ * \param[in]	name	Link name of the object. Can be NULL if parent is root,
+ *			which means operation will be on root object.
+ * \param[in]	mask	accessibility check(s) to be performed.
+ *			It should be either the value F_OK, or a mask with
+ *			bitwise OR of one or more of R_OK, W_OK, and X_OK.
+ * \param[in]	flag	bitwise flag. Handles O_NOFOLLOW.
+ *
+ * \return		0 on success, errno code on failure.
+ */
+int
+dfs_access_wflag(dfs_t *dfs, dfs_obj_t *parent, const char *name, int mask, int flag);
```

Comment:
i know i suggested this, but is this better or to reuse the libc version, like dfs_accessat ?

Reply:
In glibc,
int faccessat(int dirfd, const char *pathname, int mode, int flags);

int dfs_access(dfs_t *dfs, dfs_obj_t *parent, const char *name, int mask)
DAOS does not adopt the way of using dirfd generally. If you do prefer dfs_accessat, I can change it.

Reply:
ok no problem. this is fine as it is